MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -927,65 +934,56 @@ public IEnumerable<ExternalIdInfo> GetExternalIdInfos(IHasProviderIds item)
                 }
+
                 return dict;
             }
         }
 
         public void OnRefreshStart(BaseItem item)
         {
-            //_logger.LogInformation("OnRefreshStart {0}", item.Id.ToString("N", CultureInfo.InvariantCulture));
-            var id = item.Id;
-
-            lock (_activeRefreshes)
-            {
-                _activeRefreshes[id] = 0;
-            }
-
+            _logger.LogInformation("OnRefreshStart {0}", item.Id.ToString("N", CultureInfo.InvariantCulture));
+            _activeRefreshes[item.Id] = 0;
             RefreshStarted?.Invoke(this, new GenericEventArgs<BaseItem>(item));
         }
 
         public void OnRefreshComplete(BaseItem item)
         {
-            //_logger.LogInformation("OnRefreshComplete {0}", item.Id.ToString("N", CultureInfo.InvariantCulture));
-            lock (_activeRefreshes)
-            {
-                _activeRefreshes.Remove(item.Id);
-            }
+            _logger.LogInformation("OnRefreshComplete {0}", item.Id.ToString("N", CultureInfo.InvariantCulture));
+
+            _activeRefreshes.Remove(item.Id, out _);
 
             RefreshCompleted?.Invoke(this, new GenericEventArgs<BaseItem>(item));
         }
 
         public double? GetRefreshProgress(Guid id)
         {
-            lock (_activeRefreshes)
+            if (_activeRefreshes.TryGetValue(id, out double value))
             {
-                if (_activeRefreshes.TryGetValue(id, out double value))
-                {
-                    return value;
-                }
-
-                return null;
+                return value;
             }
+
+            return null;
         }
 
         public void OnRefreshProgress(BaseItem item, double progress)
         {
-            //_logger.LogInformation("OnRefreshProgress {0} {1}", item.Id.ToString("N", CultureInfo.InvariantCulture), progress);
             var id = item.Id;
+            _logger.LogInformation("OnRefreshProgress {0} {1}", id.ToString("N", CultureInfo.InvariantCulture), progress);
 
-            lock (_activeRefreshes)
+            if (_activeRefreshes.ContainsKey(id))
             {
-                if (_activeRefreshes.ContainsKey(id))
-                {
-                    _activeRefreshes[id] = progress;
-
-                    RefreshProgress?.Invoke(this, new GenericEventArgs<Tuple<BaseItem, double>>(new Tuple<BaseItem, double>(item, progress)));
-                }
-                else
-                {
-                    // TODO: Need to hunt down the conditions for this happening
-                    //throw new Exception(string.Format("Refresh for item {0} {1} is not in progress", item.GetType().Name, item.Id.ToString("N", CultureInfo.InvariantCulture)));
-                }
+                _activeRefreshes[id] = progress;
+                RefreshProgress?.Invoke(this, new GenericEventArgs<Tuple<BaseItem, double>>(new Tuple<BaseItem, double>(item, progress)));
+            }
```

Comment:
This is actually data race-prone, as a key might be gone between `.ContainsKey()` and assignment.